### PR TITLE
[Chore] Migrate to Xcode 12.4

### PR DESCRIPTION
**Pull request by billypchan**
## What's In This PR?

Update Azure and Jenkins to use Xcode 12.4 to build the project.

### TODO before merge
- [x] restore branches to `master`

### CODE REVIEW

File: Jenkins/UI/Build.groovy [hunk 1]
```diff
@@ -105,9 +105,11 @@ pipeline {
 		                    curl -O ${DEPENDENCIES_BASE_URL}/Gemfile.lock
 
 		                    bundle install --path ~/.gem
+                                    echo "set DEVELOPER_DIR to XCode 12.2"
+                                    export DEVELOPER_DIR=/Applications/Xcode_12.4.app/Contents/Developer
```
Comment (typfel) [inline Jenkins/UI/Build.groovy:109]: This is temporary thing you'll remove before the final merge?

File: Azure/build-and-test/carthageXcode12.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+     
+
```
Comment (typfel) [inline Azure/build-and-test/carthageXcode12.sh:2]:
```suggestion
```

**Review by johnxnguyen — COMMENTED** [review]
Could you explain a little bit what these changes are for? I'm a little confused why we're changing the branches from `master` to a chore branch, and why we want to merge this to `master`.

File: Jenkins/UI/Build.groovy (hunk 1 above)
Comment (billypchan) [reply Jenkins/UI/Build.groovy:109]: this will be restored after the default Xcode is Xcode 12.4

File: Jenkins/UI/Build.groovy [hunk 3]
```diff
@@ -105,9 +105,11 @@ pipeline {
 		                    curl -O ${DEPENDENCIES_BASE_URL}/Gemfile.lock
 
 		                    bundle install --path ~/.gem
+                                    echo "set DEVELOPER_DIR to XCode 12.2"
```
Comment (billypchan) [inline Jenkins/UI/Build.groovy:108]:
```suggestion
                                    echo "set DEVELOPER_DIR to XCode 12.4"
```

**Comment by billypchan** [comment]
I will refresh this PR which did not updated since Nov 2020

**Review by typfel — APPROVED** [review]

File: Jenkins/UI/Build.groovy (hunk 1 above)
Comment (typfel) [reply Jenkins/UI/Build.groovy]: So this is only to test the building before we've upgrade to Xcode 12.4 everywhere. That means in the only thing we'll actually merge to master is the carthage Xcode 12 workaround.

Did understand this correctly?

Comment (billypchan) [reply Jenkins/UI/Build.groovy]: yes, the branches will restore to `master` before this PR is merged.

**Review by johnxnguyen — APPROVED** [review]

File: Azure/build-and-test/build-and-test.yml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 parameters:
-  branch: "master"
+  branch: "chore/xcode12.2"
```
Comment (billypchan) [inline Azure/build-and-test/build-and-test.yml:2]:
```suggestion
  branch: "master"
```

File: Azure/framework-pipelines.yml [hunk 5]
```diff
@@ -5,7 +5,7 @@ jobs:
     steps:
     - template: build-and-test/build-and-test.yml
       parameters:
-        branch: master
+        branch: chore/xcode12
```
Comment (billypchan) [inline Azure/framework-pipelines.yml:8]:
```suggestion
        branch: master
```

File: Azure/release-framework/release-framework.yml [hunk 6]
```diff
@@ -1,12 +1,12 @@
 parameters:
-  branch: "master"
+  branch: "chore/xcode12"
```
Comment (billypchan) [inline Azure/release-framework/release-framework.yml:2]:
```suggestion
  branch: "master"
```

File: Azure/framework-pipelines.yml [hunk 7]
```diff
@@ -14,4 +14,4 @@ jobs:
     steps:
     - template: release-framework/release-framework.yml
       parameters:
-        branch: master
+        branch: chore/xcode12
```
Comment (billypchan) [inline Azure/framework-pipelines.yml:17]:
```suggestion
        branch: master
```

**Pull request merged**